Lib/test/test_builtin.py

```diff
@@ -1302,6 +1319,13 @@ def test_input(self):
             sys.stdin = io.StringIO()
             self.assertRaises(EOFError, input)
 
+            # input() in tty mode with a malformed input stream should attempt
+            # to call .readline()
+            sys.stdin = MalformedInputStream()
+            sys.stdout = MalformedOutputStream()
+            self.assertEqual(input("baz"), "foobar")  # strips \n
+            self.assertEqual(sys.stdout.value, "baz")
+
```

Comment:
Having this here is slightly confusing, as it appears like it may affect the following assertion as well. Perhaps put this at the end of this method or in a separate method.